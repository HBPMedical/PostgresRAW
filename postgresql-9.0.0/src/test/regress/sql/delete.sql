CREATE TABLE delete_test (
    id SERIAL PRIMARY KEY,
    a INT
);

INSERT INTO delete_test (a) VALUES (10);
INSERT INTO delete_test (a) VALUES (50);
INSERT INTO delete_test (a) VALUES (100);

-- allow an alias to be specified for DELETE's target table
DELETE FROM delete_test AS dt WHERE dt.a > 75;

-- if an alias is specified, don't allow the original table name
-- to be referenced
DELETE FROM delete_test dt WHERE delete_test.a > 25;

SELECT * FROM delete_test;

DROP TABLE delete_test;
