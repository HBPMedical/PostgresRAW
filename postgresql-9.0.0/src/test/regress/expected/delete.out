CREATE TABLE delete_test (
    id SERIAL PRIMARY KEY,
    a INT
);
NOTICE:  CREATE TABLE will create implicit sequence "delete_test_id_seq" for serial column "delete_test.id"
NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "delete_test_pkey" for table "delete_test"
INSERT INTO delete_test (a) VALUES (10);
INSERT INTO delete_test (a) VALUES (50);
INSERT INTO delete_test (a) VALUES (100);
-- allow an alias to be specified for DELETE's target table
DELETE FROM delete_test AS dt WHERE dt.a > 75;
-- if an alias is specified, don't allow the original table name
-- to be referenced
DELETE FROM delete_test dt WHERE delete_test.a > 25;
ERROR:  invalid reference to FROM-clause entry for table "delete_test"
LINE 1: DELETE FROM delete_test dt WHERE delete_test.a > 25;
                                         ^
HINT:  Perhaps you meant to reference the table alias "dt".
SELECT * FROM delete_test;
 id | a  
----+----
  1 | 10
  2 | 50
(2 rows)

DROP TABLE delete_test;
